### PR TITLE
Adds tests to OIDC token generation

### PR DESCRIPTION
**Pull request by Lexman42**

### CODE REVIEW

File: vault/identity_store_oidc.go [hunk 1]
```diff
@@ -352,7 +352,7 @@ func (i *IdentityStore) getOIDCConfig(ctx context.Context, s logical.Storage) (*
 	return &c, nil
 }
 
-// handleOIDCCreateKey is used to create a new named key or update an existing one
+// handleOIDCCreateUpdateKey is used to create a new named key or update an existing one
```
Comment (kalafut) [inline vault/identity_store_oidc.go]: The should be names path...

File: vault/identity_store_oidc_test.go [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+	// Read "test-role1" again
+	respReadTestRole1AfterUpdate, err2 := c.identityStore.HandleRequest(ctx, &logical.Request{
+		Path:      "oidc/role/test-role1",
+		Operation: logical.ReadOperation,
+		Storage:   storage,
+	})
+	expectSuccess(t, respReadTestRole1AfterUpdate, err2)
+
+	// Compare response for "test-role1" before and after it was updated
+	expectedDiff := map[string]interface{}{
```
Comment (kalafut) [inline vault/identity_store_oidc_test.go]: This approach seems pretty rigidly tied to `deep` (e.g. they tweak their presentation and the tests break), and I'm not sure I see the need to check what the value was. Isn't is simpler to just have an expected `map[string]interface{}` and `deep.Diff` that?

File: vault/identity_store_oidc.go [hunk 3]
```diff
@@ -585,17 +585,7 @@ func (i *IdentityStore) pathOIDCGenerateToken(ctx context.Context, req *logical.
...
-		return nil, err
-	}
-
-	te, err := i.core.LookupToken(ctx, accessorEntry.TokenID)
-	if err != nil {
-		return nil, err
-	}
-
-	if te == nil || te.EntityID == "" {
+	if req.EntityID == "" {
```
Comment (Lexman42) [inline vault/identity_store_oidc.go]: @kalafut could we just use req.EntityID here instead?

File: vault/identity_store_oidc.go [hunk 4]
```diff
@@ -607,13 +597,13 @@ func (i *IdentityStore) pathOIDCGenerateToken(ctx context.Context, req *logical.
...
 		Issuer:   config.effectiveIssuer,
-		Subject:  te.EntityID,
+		Subject:  req.EntityID,
 		Audience: role.RoleID,
 		Expiry:   now.Add(role.TokenTTL).Unix(),
 		IssuedAt: now.Unix(),
 	}
 
-	e, err := i.MemDBEntityByID(te.EntityID, true)
+	e, err := i.MemDBEntityByID(req.EntityID, true)
```
Comment (Lexman42) [inline vault/identity_store_oidc.go]: @kalafut maybe add a nil check for e? not sure what the correct logic is if it is nil. Error? Don't attempt to template groups?

File: vault/identity_store_oidc.go (hunk 3 above)
Comment (kalafut) [reply vault/identity_store_oidc.go]: yep. changed.

File: vault/identity_store_oidc.go (hunk 4 above)
Comment (kalafut) [reply vault/identity_store_oidc.go]: Yes. I'm erroring out on e==nil now.

**Comment by kalafut** [comment]
@Lexman42 Can this PR be closed now?

**Pull request closed without merging**